src/foam/dao/CachingDAO.js

```diff
@@ -108,8 +109,12 @@ foam.CLASS({
       are up to date. */
     function remove_(x, o) {
       var self = this;
-      return self.src.remove(o).then(function() {
-        return self.delegate.remove_(x, o);
+      return self.src.remove(o).then(removedObj => {
+        // only remove from cache and delegate if the object was actually removed (the removed object is not null)
+        if ( removedObj ){
+          this.onSrcRemove(removedObj);
+          return self.delegate.remove_(x, o);
```

Comment:
I don't think you need to call this.onSrcRemove since we're returning self.delegate.remove_(x, o).

Reply:
I agree.